sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,241 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements adapter for Darts models."""
+import abc
+from typing import List, Optional
+
+import pandas
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+__author__ = ["yarnabrina"]
+
+
+class _DartsAdapter(BaseForecaster):
+    """Base adapter class for Darts models.
+
+    Parameters
+    ----------
+    past_covariates : Optional[List[str]], optional
+        column names in ``X`` which are known only for historical data, by default None
+    num_samples : Optional[int], optional
+        Number of times a prediction is sampled from a probabilistic model, by default
+        1000
+
+    Notes
+    -----
+    If unspecified, all columns will be assumed to be known during prediction duration.
+    """
+
+    _tags = {
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "enforce_index_type": pandas.DatetimeIndex,
+        "handles-missing-data": False,
+        "capability:insample": False,
+        "python_version": ">=3.8",
+        "python_dependencies": ["u8darts"],
+        "python_dependecies_alias": {"u8darts": "darts"},
```

Comment:
@hazrulakmal, @fkiraly Can you please help why is this failing complaining `u8darts` is not installed, even though it is installed? Is the alias specification format incorrect?

(This is with respect to my local environment.)

Reply:
this implies that you install with `pip install u8darts`, and import with `import darts` - is that correct?

Reply:
Yes, exactly. And it's not working.

> ModuleNotFoundError: DartsXGBModel requires package 'u8darts' to be present in the python environment, but 'u8darts' was not found. 'u8darts' is a soft dependency and not included in the base sktime installation. Please run: `pip install u8darts` to install the u8darts package. To install all soft dependencies, run: `pip install sktime[all_extras]`

But I do have it.

```sh
❯ pip freeze | grep darts
u8darts==0.25.0
```

If I use `darts` as `python_dependencies`, then it starts working. But actual `darts` package in PyPI needs so much more than `u8darts` (lots of system dependencies that are not straightforward to install on all OS - as per [Darts' own documentation](https://github.com/unit8co/darts/blob/master/INSTALL.md)) that I don't want to enforce that.

Reply:
> Yes, exactly. And it's not working.

Well, that's strange.

Last person working on this and looking into the code of the `python_dependencies` handling was @hazrulakmal, I think - any ideas?

Reply:
There's a typo, it's 'python_dependencies_alias' not 'python_dependecies_alias'. That can be a reason.

Reply:
Thank you, fixed it and now we should see plenty of failures. 

But I wonder why is there no error when I used a non-existing tag. I'm not super familar with tag management system, but do we not have any check to validate if correct tags are provided or if supported valid values are provided? Can I suggest to add that enhancement?

Reply:
>  I'm not super familar with tag management system, but do we not have any check to validate if correct tags are provided or if supported valid values are provided?

That check exists, but it applies only to concrete estimators - private classes such as those starting in underscore or `Base` are excluded from the check.

The test would capture it once you inherit in a concrete class.